### PR TITLE
Upgrade Django 2.1

### PR DESCRIPTION
**Pull request by soumyaarun**
Upgraded to  Django 2.1

### CODE REVIEW

**Comment by tkdchen** [comment]
Thank you very much for making this patch.

Can you rebase and fix conflicts? You also need to:

* update tox.ini to add testenv for django 2.x. So, there will be a new testenv that is py36 and django `2.x`.
* update install_requires in setup.py to require `1.10.x`, `1.11.x` and `2.x`
* fix any code that does not work with django `2.x` and tests accordingly.

**Comment by soumyaarun** [comment]
@tkdchen,  the mentioned changes are pushed. Please verify.

File: tcms/testcases/views.py [hunk 1]
```diff
@@ -60,6 +59,12 @@
 from tcms.utils import HTTP_FORBIDDEN
 from tcms.utils import HTTP_NOT_FOUND
 
+# support django 1.1x as well as 2.x
+try:
+    from django.core.urlresolvers import reverse
+except ImportError:
+    from django.urls import reverse
+
```
Comment (tkdchen) [inline tcms/testcases/views.py]: Why try-except for importing `reverse` only here?

File: tox.ini [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
     flake8
     docs
 
 [testenv]
 deps =
     django111: Django>=1.11,<2.0
+    django200: Django>=2.0,<2.1
+    django210: Django>=2.1
+    django{111,200,210},docs: git+https://github.com/release-engineering/kobo.git#egg=kobo
+    django{111,200,210},docs: git+https://github.com/psss/python-nitrate.git#egg=nitrate
```
Comment (tkdchen) [inline tox.ini]: `python-nitrate` is not required by Nitrate actually.

File: tox.ini [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+    {py36}-django{200, 210}
     flake8
     docs
 
 [testenv]
 deps =
     django111: Django>=1.11,<2.0
+    django200: Django>=2.0,<2.1
+    django210: Django>=2.1
+    django{111,200,210},docs: git+https://github.com/release-engineering/kobo.git#egg=kobo
```
Comment (tkdchen) [inline tox.ini]: `kobo` is required by Nitrate XMLRPC, not only for building docs. Obviously, it was missed in the `install_requires`. Can you add it to that list instead?

File: tox.ini [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
     {py27,py36}-django111
+    {py36}-django{200, 210}
     flake8
     docs
 
 [testenv]
 deps =
     django111: Django>=1.11,<2.0
+    django200: Django>=2.0,<2.1
+    django210: Django>=2.1
```
Comment (tkdchen) [inline tox.ini]: I prefer `Django>=2.1,<3.0`. So, it ensures django 3.x will not be installed when it is available.

File: setup.py [hunk 5]
```diff
@@ -68,6 +67,10 @@ def get_long_description():
     ]
 }
 
+dependency_links = [
+    'https://github.com/release-engineering/kobo/tarball/master#egg=kobo',
+    'https://github.com/psss/python-nitrate/tarball/master#egg=nitrate'
```
Comment (tkdchen) [inline setup.py]: Same here, python-nitrate is not required by Nitrate.

File: tox.ini (hunk 3 above)
Comment (tkdchen) [inline tox.ini]: I see your patch for kobo. Let's wait a while for my question about the release date of kobo. Then, we can determine if we use this workaround or add kobo to `install_requires`.

**Comment by tkdchen** [comment]
Could you please sign-off your commits?

**Comment by tkdchen** [comment]
Hi @soumyaarun,

kobo 0.9.0 has been released. Can you update this PR:

* use the new kobo version and remove link `git+https://github.com/release-engineering/kobo.git#egg=kobo` from setup.py?
* fix conflicts
* sign-off commits

Thanks.

**Comment by soumyaarun** [comment]
Hi @tkdchen, 

The mentioned changes are done. Could you please verify?

**Comment by tkdchen** [comment]
Nice! Thank you very much. :+1:

**Pull request merged**